spec.bs

```diff
@@ -1036,6 +1107,24 @@ To <dfn>parse an origin</dfn> given a [=string=] |input|:
 
 </div>
 
+<div algorithm>
+
+To <dfn>get uuid from string</dfn> given a [=string=] |input|:

```

Comment:
Could we scoot this and =parse an origin= over to some helper algorithms section or something? This stuff is way too crowded...